Initial CSS module implementation

bitpshr opened the pull request:
After talking with @Tomdye last week, I worked up this PR to demonstrate how CSS modules would be used to accomplish styling for the Monster Cards app. Development felt slow and roundabout, though that could've been due to the fact that I was converting existing styles rather than implementing new styles.

Some notes:
* For now, I didn't update the sprite sheets or `global.styl`.
* New CSS module typings had to be generated every time I changed a CSS module via `dojo css typings`,
* When using [class composition](https://github.com/css-modules/css-modules#composition), a string of space-separated CSS classes is exported by webpack, causing some rather ugly template strings.

Overall, using CSS modules for a one-off application like this seemed to cause more problems than it solved, as Monster Cards already had Stylus written that had no global conflicts. I can kind of see the benefit for a set of widgets shared across many applications (even though a single, top-level unique selector on a shared widget would be sufficient 99% of the time), but for a small app like this, it seems like an unnecessary burden to impose on developers.

Refs #60

agubler commented:
@bitpshr the generated css module interfaces are causing CI to fail for not ending with a newline.

agubler commented on monster-cards/src/widgets/card/createCardSummary.ts:
```diff
@@ -2,6 +2,9 @@ import { Widget, DNode, WidgetState } from 'dojo-interfaces/widgetBases';
 import createCard from './createCard';
+import * as css from './card-summary.module.styl';
+
+console.log(css);
```
console.log

agubler commented on monster-cards/src/widgets/about/createAboutPage.ts:
```diff
@@ -13,7 +14,7 @@ const createAboutPage = createWidgetBase
 				function(this: AboutPage): DNode[] {
 					const heading = d('h1', { innerHTML: 'About' });
-					const jumbotron = d('div.jumbotron', {}, [ heading ]);
+					const jumbotron = d(`div.${ css.jumbotron.replace(/\s/g, '.') }`, {}, [ heading ]);
```
I thought we were going to write a reducer to return the correct class structure for the options?

bitpshr commented:
@Tomdye @agubler I cleaned up this PR a bit since we may land it:

- Used a reducer for composed CSS class names.
- Modularized everything but the sprite sheets and the reset. We could move these imports into `genera.module.styl` so `main.styl` has no content.
- Removed logs and addressed other PR feedback

Also note that this work depends on dojo/cli-css-typings#5 and dojo/cli-build#20 and both are ready for another review.

agubler commented:
@bitpshr This is quite a bit out of date and possibly will be superseded once the proposals that @Tomdye has been working have been implemented, so am closing.
